File: 3ra parte/flutter/flutter-end/lib/repository/api_service.dart

import 'package:chopper/chopper.dart';
import 'package:TallerUNAJ2/model/main_weather_model.dart';

part 'api_service.chopper.dart';

@ChopperApi(baseUrl: '/weather')
abstract class ApiService extends ChopperService {

  @Get()
  Future<Response> getWeatherByCity({
      @Query("q") String cityName,
      @Query("appid") String token = token,
      @Query("lang") String lang = "es"});

  static ApiService create() {
    final client = ChopperClient(baseUrl: 'http://api.openweathermap.org/data/2.5',
    services: [
      _$ApiService(),
    ],
      converter: JsonConverter(),
    );
    return _$ApiService(client);
  }

}

const String token = "42da757df1dd57f50c6c9dd0751757e3";

File: 3ra parte/kotlin/kotlin-end/app/src/main/java/com/unaj/tallerunajweatherapp/repository/networking/ApiService.kt
package com.unaj.tallerunajweatherapp.repository.networking

import com.unaj.tallerunajweatherapp.view.model.MainWeatherModel
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {
    @GET("/data/2.5/weather")
    suspend fun fetchWeatherByCity(
        @Query("q") cityName: String,
        @Query("appid") token: String,
        @Query("lang") language: String
    ): MainWeatherModel
}